aff4/data_store.h

```diff
@@ -256,6 +256,15 @@ class AFF4ScopedPtr {
         resolver_ = other.resolver_;
     }
 
+    AFF4ScopedPtr& operator=(AFF4ScopedPtr&& other) {
+        if (this != &other) {
```

Comment:
Is there any reason why this won't work instead?

```
AFF4ScopedPtr& operator=(AFF4ScopedPtr&& other) {
    if (this != &other) {
        ptr_ = other.release();
        resolver_ = other.resolver_; 
    }
    return *this; 
}
```

The nullary ctor sets both member variables to `nullptr`, so it doesn't leave any mess to deal with. The implementation of `operator=` should be just like the move ctor.

Reply:
It probably would be fine.  The pattern I use here is just a generic idiom that I've been using for copy and move operators so that you don't have to maintain the code both in the copy or move constructor and the assignment operators.  It helps to keep them from going out of sync.